diff --git a/tgcalls/platform/PlatformInterface.h b/tgcalls/platform/PlatformInterface.h
index f6434083f..515cc348f 100644
--- a/tgcalls/platform/PlatformInterface.h
+++ b/tgcalls/platform/PlatformInterface.h
@@ -88,11 +88,9 @@ public:
         return _impl->SetPlayoutDevice(index);
     }
 
-#ifdef TGCALLS_UWP_DESKTOP
     virtual int32_t SetPlayoutDevice(std::string deviceId) override {
         return _impl->SetPlayoutDevice(deviceId);
     }
-#endif
 
     virtual int32_t SetPlayoutDevice(WindowsDeviceType device) override {
         return _impl->SetPlayoutDevice(device);
@@ -102,11 +100,9 @@ public:
         return _impl->SetRecordingDevice(index);
     }
 
-#ifdef TGCALLS_UWP_DESKTOP
     virtual int32_t SetRecordingDevice(std::string deviceId) override {
         return _impl->SetRecordingDevice(deviceId);
     }
-#endif
 
     virtual int32_t SetRecordingDevice(WindowsDeviceType device) override {
         return _impl->SetRecordingDevice(device);
