diff --git a/toolchain/win/BUILD.gn b/toolchain/win/BUILD.gn
index 6b5b04fca27e..4cf4b377dce7 100644
--- a/toolchain/win/BUILD.gn
+++ b/toolchain/win/BUILD.gn
@@ -14,7 +14,8 @@ assert(is_win, "Should only be running on Windows")
 
 # Copy the VS runtime DLL for the default toolchain to the root build directory
 # so things will run.
-if (current_toolchain == default_toolchain) {
+if (current_toolchain == default_toolchain &&
+    !(target_os == "winuwp" && target_cpu == "arm")) {
   if (is_debug) {
     configuration_name = "Debug"
   } else {
